pkg/contract_watcher/transformer/transformer.go

```diff
@@ -95,40 +91,39 @@ func NewTransformer(con config.ContractConfig, bc core.BlockChain, db *postgres.
 // Loops over all of the addr => filter sets
 // Uses parser to pull event info from abi
 // Use this info to generate event filters
-func (tr *Transformer) Init() error {
+func (tr *Transformer) Init(apiKey string) error {
 	// Initialize internally configured transformer settings
-	tr.contractAddresses = make([]string, 0)       // Holds all contract addresses, for batch fetching of logs
-	tr.sortedEventIds = make(map[string][]string)  // Map to sort event column ids by contract, for post fetch processing and persisting of logs
-	tr.sortedMethodIds = make(map[string][]string) // Map to sort method column ids by contract, for post fetch method polling
-	tr.eventIds = make([]string, 0)                // Holds event column ids across all contract, for batch fetching of headers
-	tr.eventFilters = make([]common.Hash, 0)       // Holds topic0 hashes across all contracts, for batch fetching of logs
+	tr.contractAddresses = make([]string, 0)      // Holds all contract addresses, for batch fetching of logs
+	tr.sortedEventIds = make(map[string][]string) // Map to sort event column ids by contract, for post fetch processing and persisting of logs
+	tr.eventIds = make([]string, 0)               // Holds event column ids across all contract, for batch fetching of headers
+	tr.eventFilters = make([]common.Hash, 0)      // Holds topic0 hashes across all contracts, for batch fetching of logs
 	tr.Start = 100000000000
 
 	// Iterate through all internal contract addresses
 	for contractAddr := range tr.Config.Addresses {
 		// Configure Abi
 		if tr.Config.Abis[contractAddr] == "" {
 			// If no abi is given in the config, this method will try fetching from internal look-up table and etherscan
-			parseErr := tr.Parser.Parse(contractAddr)
+			parseErr := tr.Parser.Parse(contractAddr, apiKey)
 			if parseErr != nil {
-				return fmt.Errorf("error parsing contract by address: %s", parseErr.Error())
+				return fmt.Errorf("error parsing contract by address: %w", parseErr)
```

Comment:
It looks like the default apiKey is "" from the command prompt, with the user see this error then?

Reply:
No because it might succeed without the API key - seems like you only need it if you're fetching > 1 ABI